src/context-menus/users/ban.ts

```diff
@@ -0,0 +1,78 @@
+import {
+	ActionRowBuilder,
+	ApplicationCommandType,
+	ContextMenuCommandBuilder,
+	ModalBuilder,
+	TextInputBuilder,
+	TextInputStyle
+} from 'discord.js';
+import { banHandler } from '@/commands/ban';
+import type { UserContextMenuCommandInteraction, ModalActionRowComponentBuilder} from 'discord.js';
+
+export async function banContextMenuHandler(interaction: UserContextMenuCommandInteraction): Promise<void> {
+	const target = interaction.targetUser;
+
+	const reasonInput = new TextInputBuilder()
+		.setCustomId('banReason')
+		.setLabel('Reason')
+		.setRequired(true)
+		.setStyle(TextInputStyle.Short);
+
+	const deleteMessagesInput = new TextInputBuilder()
+		.setCustomId('deleteMessages')
+		.setLabel('Delete recent message history (hours)')
+		.setPlaceholder('Leave blank to not delete any messages')
+		.setRequired(false)
+		.setStyle(TextInputStyle.Short);
+
+	const reasonActionRow = new ActionRowBuilder<ModalActionRowComponentBuilder>()
+		.addComponents(reasonInput);
+
+	const deleteMessagesActionRow = new ActionRowBuilder<ModalActionRowComponentBuilder>()
+		.addComponents(deleteMessagesInput);
+
+	const banModal = new ModalBuilder()
+		.setCustomId('banModal')
+		.setTitle(`Ban ${target.tag}`)
+		.addComponents(reasonActionRow, deleteMessagesActionRow);
+
+	await interaction.showModal(banModal);
+
+	let modalSubmitInteraction;
+	try {
+		modalSubmitInteraction = await interaction.awaitModalSubmit({
+			time: 5 * 60 * 1000,
+			filter: modalSubmitInteraction =>
+				modalSubmitInteraction.customId === 'banModal' &&
+				modalSubmitInteraction.user.id === interaction.user.id
+		});
+	} catch {
+		// * The user dismissed the modal or took too long to respond
+		return;
```

Comment:
Should we be following up to the interaction here so there's some feedback about it being cancelled?

Reply:
Discord doesn't send any event when a modal is closed. That's why the time limit is necessary, so we're not waiting indefinitely. It also means that this feedback would always come after the time limit has expired, not when the modal is closed.